book/src/04_2_low_level_io.md

```diff
@@ -1,5 +1,32 @@
-# Low level I/O
+# Lower level I/O: How to manipulate Registers
+
+In general there are two ways to write firmware for the esp32. One is the bare-metal using only `[no_std]` Rust, and the other using `[std]` Rust and C-Bindings to the esp-idf.
+`[no_std]` Rust refers to Rust not using the standard library, only the core library, which is a subset of the standard library that does not depend on the existence of an operating system. 
+## What do the ecosystems look like?
+
+### `[std]` Rust and the esp-idf
+
+The most established way to use Rust on esp32-c3 is using C bindings to the esp-idf. We can use Rust's standard library when going this route, as we can use an operating system: FreeRTOS. Being able to use the standard library comes with benefits: We can use all types no matter if they are stack or heap allocated. We can use threads, Mutexes and other synchronization primitives.
```

Comment:
esp32-c3 or esp32c3? We should be consistent.

And actually I prefer _ESP32-C3_ as I think Espressif usually write it this way.